beard/metrics/text.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""String metrics.
+
+.. codeauthor:: Petros Ioannidis <petros.ioannidis91@gmail.com>
+
+"""
+
+from __future__ import division
+import re
+
+
+def find_all(s, pattern):
+    """Find all occurences of the given pattern.
+
+    Parameters
+    ----------
+    :param s: string
+        String to be searched
+
+    :param letter: string
+        Substring we are searching for
+
+    Returns
+    -------
+    :returns: generator
+        A generator that holds the indexes of the patterns
+    """
+    for match in re.finditer(pattern, s):
+        yield match.start()
+
+
+def jaro_matching(s1, s2):
+    """Return the number of matching letters.
+
+    Calculate the number of matching letters
+    based on the method for computing the Jaro distance metric.
+
+    Parameters
+    ----------
+    :param s1: string
+        First string
+
+    :param s2: string
+        Second string
+
+    Returns
+    -------
+    :returns: int
+        The number of matching letters
+    """
+    H = min(len(s1), len(s2))//2
+
+    letters_cache = dict()
+    matches = 0
+    transpositions = 0
+    s1_matching_letter_sequence = []
+    s2_matching_letter_sequence = []
+    s1_matched_positions = []
+    s2_matched_positions = []
+
+    for letter in s1:
+        if letter not in letters_cache:
+            letters_cache[letter] = (tuple(find_all(s1, letter)),
+                                     tuple(find_all(s2, letter)))
+
+    for letter, (s1_occur, s2_occur) in letters_cache.items():
```

Comment:
`items()` has different behaviour in Python2 and Python3. If you want to be consistent, use `six` library.

http://pythonhosted.org/six/#six.iteritems